### PR TITLE
Check that tag is valid semver in install.sh

### LINKED ISSUE #539: Add `version` (argument) pattern check
```
curl -sSfL https://raw.githubusercontent.com/dotenv-linter/dotenv-linter/master/install.sh | sh -s -b usr/local/bin
dotenv-linter installer: Version usr/local/bin will be installed
curl: (22) The requested URL returned error: 404
dotenv-linter installer: Failed to download file https://github.com/dotenv-linter/dotenv-linter/releases/download/usr/local/bin/dotenv-linter-darwin-x86_64.tar.gz
```

### PR DESCRIPTION
**Pull request by zd4y**
Resolves #539
Add check in install.sh to validate that the tag argument matches semver

#### ✔ Checklist:

- [x] Commit messages have been written in [Conventional Commits](https://www.conventionalcommits.org) format;

### CODE REVIEW

**Comment by mgrachev** [comment]
@zd4y Hi 👋
Thank you for your contribution!

Hmm, [linux_install check failed](https://github.com/dotenv-linter/dotenv-linter/actions/runs/3264155544/jobs/5364353765) 🤔 Could it be because of your improvements?

**Comment by zd4y** [comment]
Yes, it is giving the following warnings:
- SC3010: In POSIX sh, [[ ]] is undefined.
- SC3057: In POSIX sh, string indexing is undefined.
- SC3015: In POSIX sh, =~ regex matching is undefined.

I can fix it following the recommendations on the shellcheck wiki of each warning or changing the first line from `#!/bin/sh` to `#!/bin/bash`. Is it important to keep the script in sh?

**Comment by mgrachev** [comment]
I think yes, let's keep the script in sh.

@dotenv-linter/core Do you agree?

**Comment by mgrachev** [comment]
Please add checks to make sure that everything works.
For example: https://github.com/dotenv-linter/dotenv-linter/blob/master/.github/workflows/ci.yml#L61:L70

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,6 +69,46 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup dotenv-linter version ${{ matrix.version }}
+        shell: ${{ matrix.shell }}
+        run: |
+          mkdir -p $HOME/bin
+          sh "${GITHUB_WORKSPACE}/install.sh" -b $HOME/bin v${{ matrix.version }}
+      - name: Verify successful installation
+        shell: ${{ matrix.shell }}
+        run: test "$($HOME/bin/dotenv-linter -v)" == "dotenv-linter ${{ matrix.version }}"
```
Comment (zd4y) [inline .github/workflows/ci.yml:91]: I am new to workflows. Is it ok to use the `test` command like that there?

**Comment by mgrachev** [comment]
@zd4y Please fix the following error:
```
[Invalid workflow file: .github/workflows/ci.yml#L85](https://github.com/dotenv-linter/dotenv-linter/actions/runs/3313181346/workflow)
The workflow is not valid. .github/workflows/ci.yml (Line: 85, Col: 16): Unrecognized named-value: 'matrix'. Located at position 1 within expression: matrix.shell .github/workflows/ci.yml (Line: 90, Col: 16): Unrecognized named-value: 'matrix'. Located at position 1 within expression: matrix.shell
```

**Comment by zd4y** [comment]
Hopefully this fixes it, how can I test this in my fork?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mgrachev) [reply .github/workflows/ci.yml]: I don't see anything wrong with that 🙂

**Comment by mgrachev** [comment]
> The script not working on macos maybe has to do with `grep -P` not being available in macos? I don't have a mac so I can't test that either. I will probably try to use a different command to fix that later.

Yes, you're right. `grep` on macOS doesn't have such option:
```
$ grep -P
grep: invalid option -- P
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
```

**Comment by mgrachev** [comment]
@zd4y Is this PR ready for review?

Also, please squash your commits.

**Comment by zd4y** [comment]
> @zd4y Is this PR ready for review?
> 
> Also, please squash your commits.

Yes, please review it :)

**Review by mgrachev — APPROVED** [review]

**Comment by mgrachev** [comment]
@zd4y Thank you for your help 🔥

**Pull request merged**